src/witness_gen.rs

```diff
@@ -12,9 +13,32 @@ use trustfall::{FieldValue, TransparentValue};
 use trustfall_rustdoc::VersionedRustdocAdapter;
 
 use crate::check_release::LintResult;
+use crate::data_generation::CrateDataRequest;
 use crate::query::{Witness, WitnessQuery};
 use crate::{GlobalConfig, SemverQuery};
 
+/// Higher level data used specifically in the generation of a witness program. Values of [`None`] will
+/// prevent witness generation, since the data is required for witness generation.
+pub(crate) struct WitnessGenerationData<'a> {
+    baseline: Option<&'a CrateDataRequest<'a>>,
+    current: Option<&'a CrateDataRequest<'a>>,
+    target_dir: Option<&'a Path>,
+}
```

Comment:
Out of curiosity, would these ever realistically be `None`? At this point you probably remember this code better than I do :)

Reply:
Neat. Would you mind mentioning that in a comment?

> However, the current and baseline won't always _both_ be `None`

What if both the current and the baseline are `RustdocSource` i.e. just rustdoc JSON files? Say if `--current-rustdoc` and `--baseline-rustdoc` are used together.

Reply:
Ah, I mis-parsed your previous message. That's also what I thought. Carry on and don't mind me :)